### PR TITLE
NOVA-1783: Add swig task to transpile JSX

### PR DESCRIPTION
**Pull request by njprrogers**

### CODE REVIEW

File: lib/swig-transform-jsx/package.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    "gulp-babel": "6.x.x",
+    "gulp-tap": "0.x.x",
+    "babel-plugin-transform-react-jsx": "6.x.x",
+    "glob": "*",
+    "underscore": "*"
+  },
+  "devDependencies": {},
+  "main": "index.js",
+  "gilt": {
+    "scoped": true
```
Comment (RoryH) [inline lib/swig-transform-jsx/package.json]: This 'scoped' is no longer required.. these were added to mark a module as processed when the NPM switch happened.

File: lib/swig-transform-jsx/package.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    }
+  ],
+  "licenses": [
+    {
+      "type": "MIT",
+      "url": "https://github.com/gilt/gilt-swig-assets/blob/master/LICENSE"
+    }
+  ],
+  "engines": {
+    "node": ">= 0.12.x"
```
Comment (RoryH) [inline lib/swig-transform-jsx/package.json]: Is this correct/appropriate, do we need it even?

File: lib/swig-transform-jsx/package.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    {
+      "name": "Federico Giovagnoli",
+      "email": "fgiovagnoli@gilt.com",
+      "url": "https://github.com/Meesayen/"
+    }
+  ],
+  "licenses": [
+    {
+      "type": "MIT",
+      "url": "https://github.com/gilt/gilt-swig-assets/blob/master/LICENSE"
```
Comment (RoryH) [inline lib/swig-transform-jsx/package.json]: License in a different repo?

File: lib/swig-transform-jsx/package.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    "type": "git",
+    "url": "https://github.com/gilt/gilt-swig-assets.git"
+  },
+  "bugs": {
+    "url": "https://github.com/gilt/gilt-swig-assets/issues"
+  },
+  "author": {
+    "name": "Gilt Groupe",
+    "email": "info@gilt.com",
+    "url": "http://gilt.com/"
```
Comment (RoryH) [inline lib/swig-transform-jsx/package.json]: https 😀

File: lib/swig-transform-jsx/package.json [hunk 5]
```diff
@@ -0,0 +1,68 @@
+{
+  "name": "@gilt-tech/swig-transform-jsx",
+  "description": "Transforms JSX into Javascript.",
+  "version": "0.0.1",
+  "homepage": "https://github.com/gilt/gilt-swig-assets",
+  "keywords": [
+    "gulp",
+    "swig",
+    "package",
+    "mangement"
```
Comment (RoryH) [inline lib/swig-transform-jsx/package.json]: typo

File: lib/swig-transform-jsx/index.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  var babel = require('gulp-babel'),
+    path = require('path'),
+    tap = require('gulp-tap'),
+    basePath = path.join(swig.target.path, '/public/');
+
+  swig.tell('transform-jsx', {
+    description: 'Transforms the JSX in the source folder to javascript.'
+  });
+
+  gulp.task('transform-jsx', function () {
```
Comment (RoryH) [inline lib/swig-transform-jsx/index.js]: Should we also have a task that uses gulp.watch (IIRC) to do real-time transpiling?

**Comment by Meesayen** [comment]
Nice!

File: lib/swig-transform-jsx/package.json (hunk 2 above)
Comment (njprrogers) [inline lib/swig-transform-jsx/package.json]: it was in all the rest... again, not sure of it's usage outside of stopping pre ES6 versions of node running over the code without complaining.

File: lib/swig-transform-jsx/index.js (hunk 6 above)
Comment (njprrogers) [inline lib/swig-transform-jsx/index.js]: yes. deciding whether to use the realtime babel-core transpiler included on the page or make it a gulp task. something is coming...

**Comment by RoryH** [comment]
👍

File: lib/swig-transform-jsx/index.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    })).
+    pipe(gulp.dest(to));
+  });
+
+  gulp.task('watch-jsx', function () {
+    // Watch JS/JSX and files
+    var watchFolder = path.join(basePath, '/js/', swig.target.name, '/src/jsx/**/*.jsx');
+    gulp.watch(watchFolder, ['transform-jsx']);
+  });
+
```
Comment (njprrogers) [inline lib/swig-transform-jsx/index.js]: @RoryH Added the watcher here ^^. Might see if I can trigger it on swig run + a parameter as well.

**Comment by RoryH** [comment]
👍

File: lib/swig-transform-jsx/index.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    pipe(gulp.dest(to));
+  });
+
+
+  gulp.task('watch-jsx', function () {
+    // Watch JS/JSX and files
+    var watchFolder = path.join(basePath, '/js/', swig.target.name, '/src/jsx/**/*.jsx');
+    // if we are being invoked by swig run, only watch the folders if we have a watch-jsx parameter
+    if (_.contains(swig.argv._, 'run')) {
+      if (!swig.argv['watch-jsx']) {
```
Comment (Meesayen) [inline lib/swig-transform-jsx/index.js]: what about simply `--watch` or `--dev`?  so that we can seemingly add new watchers to `swig run` later?

**Pull request merged**